packages/cozy-authentication/src/utils/onboarding.js

```diff
@@ -0,0 +1,122 @@
+import localforage from 'localforage'
+import get from 'lodash/get'
+
+const ONBOARDING_SECRET_KEY = 'onboarding_secret'
+const ONBOARDING_STATE = 'onboarding_state'
+
+const generateRandomString = () => {
+  return Math.random()
+    .toString(36)
+    .substr(2, 11)
+}
+export const writeSecret = secret => {
+  return localforage.setItem(ONBOARDING_SECRET_KEY, secret)
+}
+
+export const readSecret = () => {
+  return localforage.getItem(ONBOARDING_SECRET_KEY)
+}
+
+export const clearSecret = () => {
+  return localforage.removeItem(ONBOARDING_SECRET_KEY)
+}
+
+export const writeState = state => {
+  return localforage.setItem(ONBOARDING_STATE, state)
+}
+
+export const readState = () => {
+  return localforage.getItem(ONBOARDING_STATE)
+}
+
+export const clearState = () => {
+  return localforage.removeItem(ONBOARDING_STATE)
+}
+const generateState = () => {
+  return generateRandomString()
+}
+const generateSecret = () => {
+  return generateRandomString()
+}
+export const checkIfOnboardingLogin = onboardingInformations => {
+  return get(onboardingInformations, 'code')
+}
+
+export const generateOAuthForUrl = async ({
+  clientName,
+  redirectURI,
+  softwareID,
+  softwareVersion,
+  clientURI,
+  logoURI,
+  policyURI,
+  scope
+}) => {
+  let secret = await readSecret()
+  if (!secret) {
+    secret = generateSecret()
+    await writeSecret(secret)
+  }
+  let state = await readState()
+
+  if (!state) {
+    state = generateState()
+    await writeState(state)
+  }
```

Comment:
Here we manage state and secret separately. Does it make sense ? It forces us to have dual methods for state/secret. Maybe they could be managed together. What happens if one exists and not the other. Is it something that is normal ? I think that if there is a discrepancy, there is a logout. If both variables form a bundle, I think it would be clearer if the code reflected that more clearly.